otherlibs/dynlink/dynlink_common.ml

```diff
@@ -354,7 +375,7 @@ module Make (P : Dynlink_platform_intf.S) = struct
   let loadfile_private filename = load true filename
 
   let unsafe_get_global_value ~bytecode_or_asm_symbol =
-    with_lock (fun () -> P.unsafe_get_global_value ~bytecode_or_asm_symbol)
+    with_lock (fun _ -> P.unsafe_get_global_value ~bytecode_or_asm_symbol)
```

Comment:
Why do we need to lock here? Is it related to the use of `Symtable` in the bytecode version?

Reply:
Yes, the bytecode version reads the `Symtable`.

Reply:
maybe a comment could be nice.

We could also pass the lock to `P.unsafe_get_global_value`. No strong opinion, I just found that it was strange to lock and then ignore the state.